tests/test_tasks.py

```diff
@@ -4226,6 +4226,35 @@ def my_flow():
         assert result == "Failed"
         assert count == 2
 
+    def test_nested_task_with_retries_on_outer_task(self):
```

Comment:
This unit test looks like what I'd have used to test, so while I didn't test myself if this work then I expect this will resolve teh issue!